common/fixation_entry.go

```diff
@@ -48,68 +48,70 @@ import (
 //
 // Thus, iterating on the prefix "packages_Entry_Index_" would yield all the package
 // indices. Reverse iterating on the prefix "packages_Entry_Raw_<INDEX>" would yield
-// all the versioned of the entry named <INDEX> in descending order.
+// all the Fixation of the entry named <INDEX> in descending order.
 //
-// 4. VersionedStore keeps a reference count of versioned of entries, and when the
+// 4. FixationStore keeps a reference count of Fixation of entries, and when the
 // count reaches 0 it marks them for deletion. The actual deletion takes place after
 // a fixed number of epochs has passed.
 
-type VersionedStore struct {
+type FixationStore struct {
 	storeKey sdk.StoreKey
 	cdc      codec.BinaryCodec
 	prefix   string
 }
 
 // AppendEntry adds a new entry to the store
-func (vs VersionedStore) AppendEntry(ctx sdk.Context, index string, block uint64, entryData codec.ProtoMarshaler) error {
+func (fs *FixationStore) AppendEntry(ctx sdk.Context, index string, block uint64, entryData codec.ProtoMarshaler) error {
 	// get the latest entry for this index
-	latestEntry := vs.getUnmarshaledEntryForBlock(ctx, index, block, types.DO_NOTHING)
+	latestEntry, err := fs.getUnmarshaledEntryForBlock(ctx, index, block, types.DO_NOTHING)
 
 	// if latest entry is not found, this is a first version entry
-	if latestEntry == nil {
-		vs.SetEntryIndex(ctx, index)
+	if latestEntry == nil && types.ErrEntryNotFound.Is(err) {
```

Comment:
i think dont return error if not found, just return nil

Reply:
if err != nil return err
else
do your thing

Reply:
done